### ISSUE TITLE
Problem with hindi text html encoding into pdf 

### ISSUE THREAD

**Issue by akashsagar8** [xhtml2pdf/xhtml2pdf#744]
### Describe the Bug
I m using xhtml2pdf Version: 0.2.5 with django version 3.2 , I m trying to convert my hindi text into pdf but after converting pdf hindi text is not encoded with a right way , it gives black boxes or unreadable characters ------

##this is my view function in django 

def show_products(request):
    products = crudajax.objects.filter(delete_flag=False).all()
    template_path = 'pdf_report.html'
    context = {'products': products}

   
    response = HttpResponse(content_type='application/pdf')
    response['Content-Disposition'] = 'attachment; filename="report.pdf"'

    template = get_template(template_path)
    html = template.render(context)
 
    result = BytesIO()
    # pisa.CreatePDF(
    #  template, dest=response,  encoding='UTF-8')
    pdf = pisa.pisaDocument(BytesIO(html.encode("UTF-16")), result, encoding='UTF-16')

   
    if not pdf.err:
      return HttpResponse(result.getvalue(), content_type='application/pdf')
    if pdf.err:
        return HttpResponse('We had some errors <pre>' + html + '</pre>')

    return response

### Expected Behavior
 I want to convert my hindi text into pdf with a encoded way 

### Actual Behavior
Actual behavior is , it convert english language only if i give another laguage like hindi , it does not encode.

### Additional Information

<details>
  <summary>Traceback</summary>

![hindi text](https://github.com/xhtml2pdf/xhtml2pdf/assets/129606427/77cea6b1-3a0b-441d-8bc0-2465744ba03c)

![hindi text](https://github.com/xhtml2pdf/xhtml2pdf/assets/129606427/656ccaef-1aab-4313-9be7-f21b03742ad8)

</details>

### System Information
I m using acer i5 

OS version:Version	10.0.22631 Build 22631

Python version:3.9.5
XHTML2PDF version:Version: 0.2.5

**Comment by stefan6419846** [comment]
Could you please provide some standalone code without relying on custom Django models and views, ideally without even using Django as this should be irrelevant in this case?

**Comment by akashsagar8** [comment]
without django also not working this code i used in a single file

**Comment by stefan6419846** [comment]
Could you please provide the standalone code you apparently have then to simplify reproduction?

**Comment by renugangwar1** [comment]
hey!!!
this is the code i am using for styling 
i have an issue when i tried to encode the hindi text it is not encoding properly 
i am using dompdf to generate pdf 
<!DOCTYPE html>
<html lang="en">

<head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0">

    <link rel="preconnect" href="https://fonts.googleapis.com">
    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
    <link
        href="https://fonts.googleapis.com/css2?family=Eczar:wght@400..800&family=Hind:wght@300;400;500;600;700&family=Noto+Sans+Devanagari:wght@100..900&family=Noto+Sans:ital,wght@0,100..900;1,100..900&display=swap"
        rel="stylesheet">


    <link rel="stylesheet" href="{{ asset('css/style.css') }}">
    <title>Enrollment Form PDF</title>
    <style>
        @font-face {
            font-family: 'NotoSansDevanagari';
            src: url('{{ public_path("assets/fonts/NotoSans_Condensed-Regular.ttf") }}') format('truetype');
            font-weight: normal;
            font-style: normal;
        }

        body {
            font-family: 'Arial', 'NotoSansDevanagari', sans-serif;
            margin: 10px;
            padding: 10px;
            font-size: 12px;
        }

        .hindi-text {
            font-family: "Hind", 'DejaVu Sans', sans-serif;
            font-weight: 400;
            font-style: normal;
        }

        .container {
            max-width: 1200px;
            /* Increase width to accommodate both form and image */
            margin: 10px auto;
            border: 1px solid #000;
            padding: 12px;
            background-color: #f9f9f9;
        }

 <div class="form-group"><label class="label">2. Name of the Student hindi:</label><span class="hindi-text">

                {{ mb_convert_encoding($enrollment->student_name_hi ?? '_________', 'UTF-8', 'auto') }}

            </span></div>

 
    }this is my controller code 
 public function downloadPDF($id)
    {
        $enrollment = Enrollment::findOrFail($id);

        if (!$enrollment) {
            return back()->with('error', 'Enrollment data not found.');
        }

        $pdf = PDF::loadView('pdf.enrollment', compact('enrollment'))->setPaper('A4');

        return $pdf->download("enrollment_form_{$id}.pdf");

![Image](https://github.com/user-attachments/assets/b63da1cd-a15e-46ed-b997-04c84f8b8456)

**Comment by PiotrCzapla** [comment]
Most likely issue with the default font here is how it is solved for slavic characters:

https://github.com/xhtml2pdf/xhtml2pdf/issues/792#issuecomment-2739895489
